src/engagement_db_to_coda/engagement_db_to_coda.py

```diff
@@ -0,0 +1,184 @@
+import json
+
+import core_data_modules.data_models
+from core_data_modules.cleaners.cleaning_utils import CleaningUtils
+from core_data_modules.logging import Logger
+from core_data_modules.util import SHAUtils, TimeUtils
+from engagement_database.data_models import HistoryEntryOrigin, MessageStatuses
+from google.cloud import firestore
+
+log = Logger(__name__)
+
+
+def _add_message_to_coda(coda, coda_dataset_config, db_message):
+    """
+    Adds a message to Coda.
+
+    If this message already has labels, copies these through to Coda.
+    Otherwise, if an auto-coder is specified, initialises with those initial labels.
+    Otherwise, adds the message with no initial labels.
+
+    :param coda: Coda instance to add the message to.
+    :type coda: coda_v2_python_client.firebase_client_wrapper.CodaV2Client
+    :param coda_dataset_config: Configuration for adding the message.
+    :type coda_dataset_config: src.engagement_db_to_coda.configuration.CodaDatasetConfiguration
+    :param db_message: Message to add to Coda.
+    :type db_message: engagement_database.data_models.Message
+    """
+    log.debug("Adding message to Coda")
+
+    coda_message = core_data_modules.data_models.Message(
```

Comment:
import Message at the top to avoid the repetition of core_data_modules.data_models

Reply:
I think this needs to be explicit, to make super clear that this is a Coda message and not an engagement db message - I'll add a comment in the code

Reply:
Actually I've changed this to `from core_data_modules.data_models import Message as CodaMessage` at the top, so we're not doing any dynamic imports